src/xmlsec/crypto.py

```diff
@@ -70,49 +75,83 @@ def from_keyspec(keyspec, private=False, signature_element=None):
 
 
 class XMlSecCrypto(object):
-    def __init__(self, source, do_padding, private):
+    def __init__(self, source, do_padding, private, do_digest=True):
         # Public attributes
         self.source = source
         self.keysize = None
         self.cert_pem = None
         self.key = None
         self.is_private = private
```

Comment:
+1

Reply:
There is a good reason for this: if you're using a PKCS11 backend you often want to use the signing function from the p11 module (i.e the priviate key) but reference the public key and certificate from outside the p11 module. The reason is that calling stuff via p11 incurs a lot of cost and you only want to spend that getting into the enclave of the p11 module.

Reply:
When I dove into it I was never really sure if a XMLSecCrypto tries to abstract a cert, a private key or a key-pair. IIRC I never saw a place where it is used as an abstraction for a keypair and I'm not sure if this would work as we only have 'self.key' as the encapsulated crypto object. rsa_x509_pem tried to abstract the difference between keys and certs away (you could just feed it any pem and it would work out on its own what it is) but still could only handle one thing in one object. cryptography strictly distinguishes between certs, pubkeys, privkeys.

My idea would be a extend XMlSecCrypto (why the lower-case "L" btw?) a bit to hold a key and a cert (and a cert-chain) at the same time and it might make the design a bit easier to understand.